src/DataGrid.tsx

```diff
@@ -203,6 +214,22 @@ function DataGrid<R, K extends keyof R, SR>({
   const [scrollTop, setScrollTop] = useState(0);
   const [scrollLeft, setScrollLeft] = useState(0);
   const [columnWidths, setColumnWidths] = useState<ReadonlyMap<string, number>>(() => new Map());
+  const [selectedPosition, setSelectedPosition] = useState<SelectCellState | EditCellState>({ idx: -1, rowIdx: -1, mode: 'SELECT' });
+  const [copiedPosition, setCopiedPosition] = useState<Position & { value: unknown } | null>(null);
+  const [isDragging, setDragging] = useState(false);
```

Comment:
If we use context then each cell needs to read the selected/copied/dragged position to set the styles. That means rerendering every cell when context value changes. I think it will be slower